src/cmark.h

```diff
@@ -329,6 +329,17 @@ CMARK_EXPORT int cmark_node_get_list_tight(cmark_node *node);
  */
 CMARK_EXPORT int cmark_node_set_list_tight(cmark_node *node, int tight);
 
+/**
+ * Returns item index of 'node'. This is only used when rendering output
+ * formats such as commonmark, which need to output the index. It is not
+ * required for formats such as html or latex.
+ */
+CMARK_EXPORT int cmark_node_get_item_index(cmark_node *node);
+
+/** Sets item index of 'node'. Returns 1 on success, 0 on failure.
+ */
+CMARK_EXPORT int cmark_node_set_item_index(cmark_node *node, int idx);
+
 /** Returns the info string from a fenced code block.
```

Comment:
Is there a reason to make these functions part of the public API?

Reply:
I was just copying the style of similar declarations in this file.

Reply:
The file `cmark.h` contains public symbols. Private symbols should be declared in one of the other header files. In this case, `node.h` seems like the best choice.

Personally, I'd simply set the struct member directly without a helper function. The whole thing is a bit of a hack because the rendering functions shouldn't modify nodes at all. But I don't see a better way to solve this problem, so I'd add a comment that the rendering code abuses the otherwise unused `list.start` member to store a temporary value.